fili-core/src/main/java/com/yahoo/bard/webservice/async/jobs/stores/ApiJobStore.java

```diff
@@ -51,4 +53,21 @@
      * @return An Observable that emits a stream of all the JobRows in the store
      */
     Observable<JobRow> getAllRows();
+
+    /**
+     * This method takes a Set of JobRowFilters, ANDS them by default, and returns a cold observable that emits a
+     * stream of JobRows which satisfy the given filter.
+     * <p>
+     * Any of the fields may be not filterable for every implementation of the {@code ApiJobStore} as the efficiency of
+     * filtering is dependent on the backing store. An IllegalArgumentException is thrown if filtering on any given
+     * field is not supported.
+     *
+     * @param jobRowFilters  A List of JobRowFilters where each JobRowFilter contains the JobField to be
+     * filtered on, the filter operation and the values to be compared to.
+     *
+     * @return An Observable that emits a stream of JobRows that satisfy the given filters
+     *
+     * @throws IllegalArgumentException if filtering on any field is not supported
+     */
+    Observable<JobRow> getFilteredRows(Set<JobRowFilter> jobRowFilters) throws IllegalArgumentException;
```

Comment:
Created a ticket for this since we will also need to add Pagination information to `getAllRows` and the focus of this PR was adding filtering functionality.

https://jira.corp.yahoo.com/browse/ADBARD-978